src/library/scala/concurrent/Future.scala

```diff
@@ -861,7 +908,7 @@ object Future {
   // by just not ever using it itself. scala.concurrent
   // doesn't need to create defaultExecutionContext as
   // a side effect.
-  private[concurrent] object InternalCallbackExecutor extends ExecutionContext with BatchingExecutor {
+  private[concurrent] final object InternalCallbackExecutor extends ExecutionContext with BatchingExecutor {
```

Comment:
Does `final object` really make a difference? Hopefully not...

Reply:
IIRC scalac automatically marks top-level objects `final`, so this should be redundant.
Nested objects however, whether `case` or not, aren't marked `final`, which is why we started marking all our `case object`s as `final` as well, which also nicely aligns them with all `case class`es, which we also always mark as `final` for some time now...

Reply:
It's an object within an object and I don't take chances. :)